private-templates-service/client/src/store/currentTemplate/actions.ts

```diff
@@ -0,0 +1,95 @@
+    error: error
+  };
+}
+
+function requestExistingTemplateUpdate(): CurrentTemplateAction {
+  return {
+    type: REQUEST_EXISTING_TEMPLATE_UPDATE,
+    text: "requesting post existing template on save"
+  };
+}
+
+function receiveExistingTemplateUpdate(templateJSON: string, templateName: string, sampleDataJSON: string): CurrentTemplateAction {
+  return {
+    type: RECEIVE_EXISTING_TEMPLATE_UPDATE,
+    text: "receiving post existing template on save",
+    templateJSON: templateJSON,
+    templateName: templateName,
+    sampleDataJSON: sampleDataJSON
+  };
+}
+
+function failureExistingTemplateUpdate(error: IncomingMessage): CurrentTemplateAction {
+  return {
+    type: FAILURE_EXISTING_TEMPLATE_UPDATE,
+    text: "failure post existing template on save",
+    error: error
+  };
+}
+
+export function updateTemplate(templateID: string, templateJSON: string, sampleDataJSON: string, templateName: string) {
+  return function (dispatch: any) {
+    let api = new TemplateApi();
+    api.setApiKey(0, "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6IkhsQzBSMTJza3hOWjFXUXdtak9GXzZ0X3RERSJ9.eyJhdWQiOiI4NjMxY2E0ZS1hMjVkLTQ5YWUtYmQ5OC1mZjNlMWRkZDQ0MTgiLCJpc3MiOiJodHRwczovL2xvZ2luLm1pY3Jvc29mdG9ubGluZS5jb20vNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3L3YyLjAiLCJpYXQiOjE1ODEzNjA2MTgsIm5iZiI6MTU4MTM2MDYxOCwiZXhwIjoxNTgxMzY0NTE4LCJhaW8iOiI0Mk5nWUdDVzk3Qm1NamRrY3pQK1pLSGZMeFlPQUE9PSIsImF6cCI6Ijg2MzFjYTRlLWEyNWQtNDlhZS1iZDk4LWZmM2UxZGRkNDQxOCIsImF6cGFjciI6IjEiLCJvaWQiOiI4YWRiZWRhNi00ZDdkLTQ1N2ItOTNkZC1jMTVmMWE1Y2NiMzEiLCJzdWIiOiI4YWRiZWRhNi00ZDdkLTQ1N2ItOTNkZC1jMTVmMWE1Y2NiMzEiLCJ0aWQiOiI3MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDciLCJ1dGkiOiJ6SDlkcHVQNnprcUJidC1MUnVGYkFBIiwidmVyIjoiMi4wIn0.WhdvkqeJymW-Ayeht6tafd8Z1muoMnyPhKoYq6KGrHdv6psfYQtmK0P0-TA5_zgOOHNNJvpVqH2LPnZTbpK4qgKLByR9umELHgD2FW9v5Djg1NAKqmQEGg_-Th__SGE3L9_WI58Wh0_Toh3f7fpLDzNBiC5iYDdGSaTilwxaYMGbXnJO2Y6Tow83GQATKGA3B27Xz2iBO9UFmEy9rte4DyLUAEIE6SCKwg-3YuD0zKfpgyd-lvhZZr37HeE9Y6hyOm_M4b_jwWI7oK0uZASuQer83W5jsSZNtiXg_T0euXJcvI9lL6R4oJDI_N6Y42RdDioIwX6FzOA4vRzTySZjBw");
+
+    let newTemplate = new TemplateJSON();
+    newTemplate.template = templateJSON;
+    newTemplate.name = templateName;
+    newTemplate.isPublished = false;
+
+    if (templateID === null || templateID === "") {
+      dispatch(requestNewTemplateUpdate());
+      return api.templatePost(newTemplate).then(response => {
+        if (response.response.statusCode && response.response.statusCode === 201 && response.body.id) {
+          dispatch(receiveNewTemplateUpdate(response.body.id, templateJSON, templateName, sampleDataJSON));
+        } else {
+          dispatch(failureNewTemplateUpdate(response.response));
+        }
+      });
+    }
+    else {
+      dispatch(requestExistingTemplateUpdate());
+      return api.templateTemplateIdPost(templateID, newTemplate).then(response => {
+        if (response.response.statusCode && response.response.statusCode === 201) {
```

Comment:
If updating an existing resource, is the status code 200 or 201?

Reply:
I think the way the endpoint was setup it returns a 201 on success, but I can see convention says it should be 200. We can address this separately.

Reply:
okay!